Library/Homebrew/rubocops/uses_from_macos.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require "rubocops/extend/formula"
+
+module RuboCop
+  module Cop
+    module FormulaAudit
+      # This cop audits `uses_from_macos` dependencies in formulae
+      class UsesFromMacos < FormulaCop
+        ALLOWED_USES_FROM_MACOS_DEPS = %w[
+          bison
+          bzip2
+          curl
+          expat
+          expect
+          flex
+          groff
+          libffi
+          libxml2
+          libxslt
+          ncurses
+          m4
+          perl
+          php
+          python@2
+          ruby
+          sqlite
+          texinfo
+          unzip
+          vim
+          xz
+          zlib
+          zsh
+        ].freeze
```

Comment:
```suggestion
        ALLOWED_USES_FROM_MACOS_DEPS = %w[
          bison
          bzip2
          curl
          expat
          expect
          flex
          groff
          icu4c
          krb5
          libedit
          libffi
          libpcap
          libxml2
          libxslt
          llvm
          ncurses
          m4
          openldap
          openssl
          perl
          php
          ruby
          sqlite
          tcl-tk
          texinfo
          unzip
          vim
          xz
          zlib
          zip
          zsh
        ].freeze
```

I checked all the failures in homebrew/core and have added the valid ones here.